### PR TITLE
[rush] Fix an issue where parameter values containing spaces aren't correctly passed to global commands.

### PR DESCRIPTION
**Pull request by iclanton**
## Summary

Fixes https://github.com/microsoft/rushstack/issues/3027

## Details

Escapes quotemarks in args and wraps args containing spaces in quotemarks before passing them to the shell.

## How it was tested

Tested with https://github.com/iclanton/rushstack/commit/44c1948519788c7b13b21939c5402473623ef2da and running `rush my-global-command --my-string "x \"y z"`

### CODE REVIEW

File: libraries/ts-command-line/src/test/__snapshots__/CommandLineParameter.test.ts.snap [hunk 1]
```diff
@@ -556,15 +644,15 @@ a longer description
 
 Optional arguments:
   -h, --help            Show this help message and exit.
-  -c {one,two,three,default}, --choice {one,two,three,default}
+  -c {one,two,three,choice with spaces,default}, --choice {one,two,three,choice with spaces,default}
```
Comment (dmichon-msft) [inline libraries/ts-command-line/src/test/__snapshots__/CommandLineParameter.test.ts.snap:647]: The help docs should really be quoting anything that needs to be quoted.

File: common/reviews/api/ts-command-line.api.md [hunk 2]
```diff
@@ -30,7 +30,7 @@ export class CommandLineChoiceListParameter extends CommandLineParameter {
     constructor(definition: ICommandLineChoiceListDefinition);
     readonly alternatives: ReadonlyArray<string>;
     // @override
-    appendToArgList(argList: string[]): void;
+    appendToArgList(argList: string[], wrapValueInQuotesIfNecessary?: boolean): void;
```
Comment (dmichon-msft) [inline common/reviews/api/ts-command-line.api.md:33]: I think quoting values should ultimately be the responsibility of the entity using the argument list. It's a little weird to have it be the responsibility of the parser when the thing consuming the argument list can easily quote the arguments with a single call to `.map()`.

Comment (iclanton) [reply common/reviews/api/ts-command-line.api.md:33]: Fair enough.

File: libraries/ts-command-line/src/test/__snapshots__/CommandLineParameter.test.ts.snap (hunk 1 above)
Comment (iclanton) [reply libraries/ts-command-line/src/test/__snapshots__/CommandLineParameter.test.ts.snap:647]: We should fix that separately.

**Review by dmichon-msft — APPROVED** [review]

File: apps/rush-lib/src/cli/scriptActions/GlobalScriptAction.ts [hunk 3]
```diff
@@ -104,6 +104,17 @@ export class GlobalScriptAction extends BaseScriptAction {
       customParameter.appendToArgList(customParameterValues);
     }
 
+    for (let i: number = 0; i < customParameterValues.length; i++) {
+      let customParameterValue: string = customParameterValues[i];
+      customParameterValue = customParameterValue.replace(/"/g, '\\"');
+
+      if (customParameterValue.indexOf(' ') >= 0) {
```
Comment (D4N14L) [inline apps/rush-lib/src/cli/scriptActions/GlobalScriptAction.ts:111]: Are there any other reasons we would want to wrap the provided string in quotes (special chars?)

**Review by D4N14L — APPROVED** [review]

File: apps/rush-lib/src/cli/scriptActions/GlobalScriptAction.ts (hunk 3 above)
Comment (iclanton) [reply apps/rush-lib/src/cli/scriptActions/GlobalScriptAction.ts:111]: I'm pretty sure spaces are the only characters that matter.

**Pull request merged**